### PR TITLE
include nbformat in `install_requires` as dependency to fix #692

### LINKED ISSUE #692: jsonschema.SchemaError when  a figure  is validated
Here is a minimal example that reproduces the bug: http://nbviewer.jupyter.org/gist/empet/cf922d7c7f4269d6f63432ec67a5d020

The notebook runs OK (with plotly 2.0.1) when I call `plot(fig)`. `iplot(fig)` generates the plot too, but an error box pops up whenever Jupyter tries to save the notebook.  The box has the following content:

_The save operation succeeded, but the notebook does not appear to be valid. The validation error was:
Notebook Validation failed_:
`u'data': [{u'colorscale': u'Viridis', u'z': [[2, 27, 105, 100], [87, 14, 121, 102], [26, 121, 73, 34], [44, 105, 111, 127]], u'type': u'heatmap', u'zsmooth': u'best'}], u'layout': {u'width': 400, u'height': 400}}` _is not valid under any of the given schemas_:

`{
 "data": [
  {
   "colorscale": "Viridis",
   "z": [
    [
     2,
     27,
     105,
     100
    ],
    [
     87,
     14,
     121,
     102
    ],
    [
     26,
     121,
     73,
     34
    ],
    [
     44,
     105,
     111,
     127
    ]
   ],
   "type": "heatmap",
   "zsmooth": "best"
  }
 ],
 "layout": {
  "width": 400,
  "height": 400
 }
}`

Initially I formulated this issue only for heatmaps, but meanwhile I realized that this behaviour manifests for any type of  plot.

### PR DESCRIPTION
**Pull request by chriddyp**
From https://github.com/plotly/plotly.py/issues/692#issuecomment-281170122

I have not tested this locally.

cc @theengineear @cldougl @rgbkrk

### CODE REVIEW

**Comment by chriddyp** [comment]
also cc @Kully

File: setup.py [hunk 1]
```diff
@@ -45,5 +45,9 @@ def readme():
                 'plotly/matplotlylib/mplexporter',
                 'plotly/matplotlylib/mplexporter/renderers'],
       package_data={'plotly': ['package_data/*']},
-      install_requires=['decorator', 'requests', 'six', 'pytz'],
+      install_requires=['decorator',
+                        'nbformat>=4.3',
```
Comment (cldougl) [inline setup.py]: @chriddyp should this be 4.3? upgrading I got only 4.2

**Comment by theengineear** [comment]
I'm gonna defer to @cldougl since I can't get this to fail when I'm using `nbformat===4.1.0` on my machine.

File: setup.py (hunk 1 above)
Comment (chriddyp) [inline setup.py]: Oh ok, yeah, 4.2 should be good. I see `mime-type` listed here: https://github.com/jupyter/nbformat/blob/master/docs/changelog.rst. Earlier versions might be OK too

Comment (cldougl) [inline setup.py]: oh I see -- it's not on pip yet https://github.com/jupyter/nbformat/issues/90
so our tests were failing with 
```
Collecting nbformat>=4.3 (from plotly==2.0.1)
  Could not find a version that satisfies the requirement nbformat>=4.3 (from plotly==2.0.1) (from versions: 4.0.0, 4.0.1, 4.1.0, 4.2.0)
No matching distribution found for nbformat>=4.3 (from plotly==2.0.1)
```
https://circleci.com/gh/plotly/plotly.py/1942?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by cldougl** [comment]
💃

**Pull request merged**